assets/app/reducers.js

```diff
@@ -39,36 +39,21 @@ export function builds(state = [], action) {
 export function sites(state = [], action) {
   switch (action.type) {
     case siteActionTypes.SITES_RECEIVED:
-      return action.sites || []
+      return action.sites || [];
     case siteActionTypes.SITE_ADDED:
       // if a site hasn't been properly added, return the existing state
       // TODO: why is this getting called if there is an error
       return action.site ? [...state, action.site] : state;
     case siteActionTypes.SITE_UPDATED:
-      return state.map((site) => {
-        if (site.id !== action.siteId) return site;
-
-        return Object.assign({}, site, action.site);
-      });
-    case siteActionTypes.SITE_DELETED:
-      return state.filter((site) => site.id != action.siteId);
+      return mapPropertyToMatchingSite(state, action.siteId, action.site);
     case siteActionTypes.SITE_CONFIGS_RECEIVED:
-      return state.map((s) => {
-        if (s.id !== action.siteId) return s;
-        return Object.assign({}, s, action.configs);
-      });
+      return mapPropertyToMatchingSite(state, action.siteId, action.configs);
     case siteActionTypes.SITE_ASSETS_RECEIVED:
-      return state.map((s) => {
-        if (s.id !== action.siteId) return s;
-        return Object.assign({}, s, { assets: action.assets });
-      });
+      return mapPropertyToMatchingSite(state, action.siteId, { assets: action.assets });
     case siteActionTypes.SITE_CONTENTS_RECEIVED:
-      return state.map((s) => {
-        if (s.id !== action.siteId) return s;
-        return Object.assign({}, s, {
-          files: action.files
-        });
-      });
+      return mapPropertyToMatchingSite(state, action.siteId, { files: action.files });
+    case siteActionTypes.SITE_DELETED:
+      return state.filter((site) => site.id != action.siteId);
     case siteActionTypes.SITE_CHILD_CONTENT_RECEIVED:
       const currentMap = state.childDirectoriesMap || {};
```

Comment:
Yes, exactly!

On Thu, Jul 21, 2016 at 8:06 AM, Brandon Kirby <notifications@github.com>
wrote:

> In assets/app/reducers.js
> <https://github.com/18F/federalist/pull/467#discussion_r71723668>:
>
> >      case siteActionTypes.SITE_CONTENTS_RECEIVED:
> > -      return state.map((s) => {
> > -        if (s.id !== action.siteId) return s;
> > -        return Object.assign({}, s, {
> > -          files: action.files
> > -        });
> > -      });
> > +      return mapPropertyToMatchingSite(state, action.siteId, { files: action.files });
> > +    case siteActionTypes.SITE_DELETED:
> > +      return state.filter((site) => site.id != action.siteId);
> >      case siteActionTypes.SITE_CHILD_CONTENT_RECEIVED:
> >        const currentMap = state.childDirectoriesMap || {};
>
> Was really meant to be this:
>
>     case siteActionTypes.SITE_CHILD_CONTENT_RECEIVED:
>       return state.map((site) => {
>         if (site.id !== action.siteId) return site;
>
>         const currentMap = site.childDirectoriesMap || {};
>         const nextMap = {};
>         nextMap[action.path] = action.files || [];
>
>         return Object.assign({}, site, {
>           childDirectoriesMap: Object.assign({}, currentMap, nextMap)
>         });
>       });
>
> ?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/18F/federalist/pull/467/files/a263a8df5d37bd5eb92498d3155d00eed43cef36#r71723668>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABWyGGbR5wmbr5Cp2fKBYt6i5jB1dM-vks5qX4sDgaJpZM4JQ3fJ>
> .
>